File: 1_arrays/1_set_matrix_zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        r: set[int] = set()
        c: set[int] = set()
        m, n = len(matrix), len(matrix[0])
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 0:
                    r.add(i)
                    c.add(j)
        for i in range(m):
            for j in range(n):
                if (i in r) or (j in c):
                    matrix[i][j] = 0

File: 01_arrays/1_set_matrix_zeroes.rs
use std::collections::HashSet;

impl Solution {
    pub fn set_zeroes(matrix: &mut Vec<Vec<i32>>) {
        let mut r: HashSet<usize> = HashSet::new();
        let mut c: HashSet<usize> = HashSet::new();

        let n = matrix.len();
        let m = matrix[0].len();

        for i in 0..n {
            for j in 0..m {
                if matrix[i][j] == 0 {
                    r.insert(i);
                    c.insert(j);
                }
            }
        }

        for i in 0..n {
            for j in 0..m {
                if r.contains(&i) || c.contains(&j) {
                    matrix[i][j] = 0;
                }
            }
        }
    }
}